jobs/populate_db.py

```diff
@@ -28,65 +41,57 @@ def insert_into_db(data_set):
+            2a. Pass the list of empty fields to add_fields
+            2b. Pass the list of filled fields to remove_fields
 
-        Stage 3: Run sort_fields (this sorts the annotations fields into two categories: fields that have entries, and fields that do not)
-            3a. Pass the list of empty fields to add_fields
-            3b. Pass the list of filled fields to remove_fields
+        Stage 3: Add/remove tasks as necessary.
+            3a. add_fields will check to see if a task exists and add one if needed
+            3b. remove_fields will check to see if an incomplete task exists, and will remove it if it does
 
-        Stage 4: Add/remove tasks as necessary.
-            4a. add_fields will check to see if a task exists and add one if needed
-            4b. remove_fields will check to see if an incomplete task exists, and will remove it if it does
-        At this point the process is complete and the next tool is passed to check_for_entry.
+        At this point the process is complete and the next tool is passed to add_or_update_tool.
     """
+    logger.info("Begin tool processing and db insertion.")
     for tool in data_set:
-        check_for_entry(tool)
-    return "All tools have been processed."
+        add_or_update_tool(tool)
+    logger.info("Tool processing complete.")
+    return
 
 
-def check_for_entry(tool):
-    """Receives a tool and checks to see if an entry exists in the DB."""
-    tool_name = tool["name"]
-    result = db.session.execute(select(Tool).where(Tool.name == tool_name)).all()
-    # If an entry exists move on to the deprecation check
-    if len(result) > 0:
-        print(f"{tool_name} already exists in database.")
+def add_or_update_tool(tool):
+    """Receives a tool and adds an entry to the tool table."""
+    insert_stmt = insert(Tool).values(
+        name=tool["name"],
+        title=tool["title"],
+        description=tool["description"],
+        url=tool["url"],
+    )
+    on_duplicate_insert_stmt = insert_stmt.on_duplicate_key_update(
+        title=tool["title"], description=tool["description"], url=tool["url"]
+    )
+    db.session.execute(on_duplicate_insert_stmt)
+    db.session.commit()
+    if is_deprecated(tool):
+        handle_deprecation(tool)
     else:
-        add_tool_entry(tool)
-    check_deprecation(tool)
+        sort_fields(tool)
 
 
-def add_tool_entry(tool):
-    """Receives a tool and adds an entry to the tool table."""
-    tool = {
-        "name": tool["name"],
-        "title": tool["title"],
-        "description": tool["description"],
-        "url": tool["url"],
-    }
-    db.session.execute(insert(Tool), tool)
-    db.session.commit()
-    print(f"{tool['name']} inserted into db")
+def is_deprecated(tool):
+    """Returns a Boolean indicating if the tool is deprecated."""
+    return tool["deprecated"] is True or tool["annotations"]["deprecated"] is True
```

Comment:
This can be simplified to `return tool["deprecated"] or tool["annotations"]["deprecated"]`